test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -0,0 +1,246 @@
+{
+	return _os << "(" << _x.val() << ")";
+}
+
+ostream& operator<<(ostream& _os , VarRef const& _x)
+{
+	return _os  << "x_" << (static_cast<uint32_t>(_x.varnum()) % 10);
+}
+
+ostream& operator<<(ostream& _os , Expression const& _x)
+{
+	if (_x.has_varref())
+		return _os  << _x.varref();
+	else if (_x.has_cons())
+		return _os  << _x.cons();
+	else if (_x.has_binop())
+		return _os  << _x.binop();
+	else
+		return _os  << "1";
+}
+
+ostream& operator<<(ostream& _os , BinaryOp const& _x)
+{
+	switch (_x.op()) {
+		case BinaryOp::ADD:
+			_os << "add(";
+			break;
+		case BinaryOp::SUB:
+			_os << "sub(";
+			break;
+		case BinaryOp::MUL:
+			_os << "mul(";
+			break;
+		case BinaryOp::DIV:
+			_os << "div(";
+			break;
+		case BinaryOp::MOD:
+			_os << "mod(";
+			break;
+		case BinaryOp::XOR:
+			_os << "xor(";
+			break;
+		case BinaryOp::AND:
+			_os << "and(";
+			break;
+		case BinaryOp::OR:
+			_os << "or(";
+			break;
+		case BinaryOp::EQ:
+			_os << "eq(";
+			break;
+		case BinaryOp::LT:
+			_os << "lt(";
+			break;
+		case BinaryOp::GT:
+			_os << "gt(";
+			break;
+	}
+	return _os << _x.left() << "," << _x.right() << ")";
+}
+
+// New var numbering starts from x_10 until x_16
+ostream& operator<<(ostream& _os , VarDecl const& _x)
+{
+	_os << "let x_" << ((_x.id() % 7) + 10) << " := " << _x.expr() << "\n";
+	return _os;
+}
+
+ostream& operator<<(ostream& _os , TypedVarDecl const& _x)
+{
+	_os << "let x_" << ((_x.id() % 7) + 10);
+	switch (_x.type())
+	{
+		case TypedVarDecl::BOOL:
+			_os << ": bool := " << _x.expr() << " : bool\n";
+			break;
+		case TypedVarDecl::S8:
+			_os << ": s8 := " << _x.expr() << " : s8\n";
+			break;
+		case TypedVarDecl::S32:
+			_os << ": s32 := " << _x.expr() << " : s32\n";
+			break;
+		case TypedVarDecl::S64:
+			_os << ": s64 := " << _x.expr() << " : s64\n";
+			break;
+		case TypedVarDecl::S128:
+			_os << ": s128 := " << _x.expr() << " : s128\n";
+			break;
+		case TypedVarDecl::S256:
+			_os << ": s256 := " << _x.expr() << " : s256\n";
+			break;
+		case TypedVarDecl::U8:
+			_os << ": u8 := " << _x.expr() << " : u8\n";
+			break;
+		case TypedVarDecl::U32:
+			_os << ": u32 := " << _x.expr() << " : u32\n";
+			break;
+		case TypedVarDecl::U64:
+			_os << ": u64 := " << _x.expr() << " : u64\n";
+			break;
+		case TypedVarDecl::U128:
+			_os << ": u128 := " << _x.expr() << " : u128\n";
+			break;
+		case TypedVarDecl::U256:
+			_os << ": u256 := " << _x.expr() << " : u256\n";
+			break;
+	}
+	return _os;
+}
+
+ostream& operator<<(ostream& _os , UnaryOp const& _x)
+{
+	switch (_x.op()) {
+		case UnaryOp::NOT:
+			_os << "not(";
+			break;
+		case UnaryOp::MLOAD:
+			_os << "mload(";
+			break;
+		case UnaryOp::SLOAD:
+			_os << "sload(";
+			break;
+		case UnaryOp::ISZERO:
+			_os << "iszero(";
+			break;
+	}
+	return _os  << _x.operand() << ")";
+}
+
+ostream& operator<<(ostream& _os , AssignmentStatement const& _x)
+{
+	return _os  << _x.ref_id() << " := " << _x.expr() << "\n";
```

Comment:
Don't you need `x_`? And what abouth the `% 10` operations you did for the declarations?

Reply:
No, `_x.ref_id()` returns a `VarRef` that then prints out `x_` and `%` that you refer to.